tests/scripts/all.sh

```diff
@@ -127,6 +129,7 @@ General options:
      --armcc            Run ARM Compiler builds (on by default).
      --no-armcc         Skip ARM Compiler builds.
      --no-yotta         Skip yotta module build.
+     --no-cxx           Skip CXX Compiler build.
```

Comment:
Why might someone want to skip the C++ build? I don't see a reason for having an option.

Reply:
`make` shouldn't run C++ by default because there are C compilers without an associated C++ compilers. But `all.sh` assumes that you have GCC, which has a C++ compiler. So I agree, this doesn't need to be optional in `all.sh`.

Reply:
I'd rather assume that there are targets that don't have a C++ compiler in the toolchain. It may happen for some embedded targets that have their toolchains generated (and stripped off any unnecessary tools). I just wanted to make it possible for these people to turn the support off with ease, hoping that giving more possibilities is better. This test will run by default by the way.

Reply:
As Gilles said, this would be a relevant argument if we were talking about `make`, but here we're talking about `all.sh`, which doesn't build for arbitrary targets: it builds for a very specific set of targets with a very specific set of toolchains (namely gcc, clang, arm-gcc, armcc5 and armclang). And we specify the exact toolchain on each `make` invocation. This isn't a general-purpose script, it already requires a very specific environment (in addition to the specific toolchains, specific version of OpenSSL and GnuTLS, and so on), and is not meant to be adapted to everyone's needs: it's just for us and the CI.

We should take advantage of the fact that this script is not generic and avoid any unnecessary complexity.